src/Native/System.Native/pal_runtimeinformation.cpp

```diff
@@ -26,14 +26,36 @@ extern "C" int32_t SystemNative_GetUnixVersion(char* version, int* capacity)
 /* Returns an int representing the OS Architecture:
  0 - x86
  1 - x64
- 2 - ARM */
-extern "C" int32_t SystemNative_GetUnixArchitecture()
+ 2 - ARM
+ 3 - ARM64 */
+extern "C" int32_t SystemNative_GetOSArchitecture()
 {
-#if defined(ARM)
+#if defined(_ARM_)
     return ARCH_ARM;
-#elif defined(X64)
+#elif defined(_ARM64_)
+    return ARCH_ARM64;
+#elif defined(_AMD64_)
     return ARCH_X64;
-#elif defined(X86)
+#elif defined(_X86_)
+    return ARCH_X86;
+#error Unidentified Architecture
+#endif
+}
+
+/* Returns an int representing the OS Architecture:
+0 - x86
+1 - x64
+2 - ARM
+3 - ARM64 */
+extern "C" int32_t SystemNative_GetProcessArchitecture()
+{
+#if defined(_ARM_)
+    return ARCH_ARM;
+#elif defined(_ARM64_)
+    return ARCH_ARM64;
+#elif defined(_AMD64_)
+    return ARCH_X64;
+#elif defined(_X86_)
     return ARCH_X86;
 #error Unidentified Architecture
```

Comment:
Although functionally it is fine here, since we return from the other if conditions before. Semantically same meaning, but not sure why compiler didn't complain